packages/cpp/src/node/cpp-contribution.ts

```diff
@@ -29,6 +29,7 @@ export class CppContribution extends BaseLanguageServerContribution {
             if (message.method === InitializeRequest.type.method) {
                 const initializeParams = message.params as InitializeParams;
                 initializeParams.processId = process.pid;
+                initializeParams.initializationOptions = this.cppPreferences['cpp.indexingExclusions'];
```

Comment:
I think you might need to introduce the "ClangdConfigurationParamsChange" interface on the client side too. Because here, it should be initializeParams.initializationOptions.ExclusionList = this.cppPreferences['cpp.indexingExclusions'];

Reply:
initializationOptions is defined as 'any' in LSP, so I'm not even sure how we would go about that on the client side.

Reply:
I mean, make a new interface in the client side and use that instead of "any".
```
interface ClangdConfigurationParamsChange {
    ExclusionList?: string[]
}
```